### PR TITLE
feat(backend): creating migration files

### PR DESCRIPTION
**Pull request by domin191013**
### Context
- db migration files creation

### Changes
Added migration files for different tables

### CODE REVIEW

File: packages/backend/src/database/knexfile.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+module.exports = {
```
Comment (beniaminmunteanu) [inline packages/backend/src/database/knexfile.js:1]: I know that you used what i had on my draft PR, but upon second thought, I now think that the knex config and migrations should be moved out of src.

Let's move the knexfile.js in packages/backend, at the same level with src.
Also move the migrations directory in packges/backend at the same level with src.

File: packages/backend/package.json [hunk 2]
```diff
@@ -1,14 +1,34 @@
 {
   "name": "backend",
   "scripts": {
-    "build": "tsc --build tsconfig.json"
+    "build": "tsc --build tsconfig.json",
+    "knex": "./node_modules/.bin/knex --knexfile src/database/knexfile.js"
```
Comment (beniaminmunteanu) [inline packages/backend/package.json:5]: if the knex file is moved as per my previous comment of this review, I think we can then just have this script as:
"kenx": knex

File: packages/backend/migrations/20230209092730_account.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.createTable('accounts', (table) => {
+    table.string('rapyd_account_id').primary()
+    table.string('user_id')
+    table.string('asset_code')
+
+    table.timestamps(false, true)
```
Comment (raducristianpopa) [inline packages/backend/migrations/20230209092730_account.js:11]:
```suggestion
    table.uuid('id').notNullable().primary()
    table.string('rapyd_account_id').notNullable()
    table.string('user_id').notNullable()
    table.string('asset_code').notNullable()

    table.timestamps(false, true, true)
```

File: packages/backend/migrations/20230209095515_user.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  return knex.schema.createTable('users', (table) => {
+    table.increments('id').primary()
+    table.string('username')
+    table.string('email')
+    table.string('password')
+    table.string('last_name')
+    table.string('first_name')
+    table.string('rapyd_wallet_id')
+    table.string('rapyd_contact_id')
+    table.timestamps(false, true)
```
Comment (raducristianpopa) [inline packages/backend/migrations/20230209095515_user.js:15]:
```suggestion
    table.uuid('id').notNullable().primary()
    table.string('email').notNullable()
    table.string('password').notNullable()
    table.string('last_name').notNullable()
    table.string('first_name').notNullable()
    table.string('rapyd_wallet_id').notNullable()
    table.string('rapyd_contact_id').notNullable()
    
    table.timestamps(false, true, true)
```

File: packages/backend/migrations/20230209095526_payment_pointer.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.createTable('payment_pointers', (table) => {
+    table.string('payment_pointer_id').primary()
+    table.integer('account_id')
+
+    table.timestamps(false, true)
```
Comment (raducristianpopa) [inline packages/backend/migrations/20230209095526_payment_pointer.js:10]:
```suggestion
    table.uuid('id').notNullable().primary()
    table.uuid('account_id').notNullable()
    table.foreign('account_id').references('accounts.id')

    table.timestamps(false, true, true)
```

File: packages/backend/migrations/20230209095548_grant.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.createTable('grants', (table) => {
+    table.increments('id').primary()
+
+    table.timestamps(false, true)
```
Comment (raducristianpopa) [inline packages/backend/migrations/20230209095548_grant.js:9]:
```suggestion
    table.uuid('id').notNullable().primary()

    table.timestamps(false, true, true)
```

File: packages/backend/migrations/20230209130752_transaction.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    table.increments('id').primary()
+    table.integer('payment_id')
+    table.string('description')
+    table.integer('payment_pointer_id')
+    table.string('asset_code')
+    table.float('value')
+    table.enum('type', ['INCOMING', 'OUTGOING']).notNullable()
+    table.enum('status', ['PENDING', 'COMPLETED', 'REJECTED']).notNullable()
+
+    table.timestamps(false, true)
```
Comment (raducristianpopa) [inline packages/backend/migrations/20230209130752_transaction.js:16]:
```suggestion
    table.uuid('id').notNullable().primary()
    table.uuid('payment_id').notNullable()
    table.string('description')
    
    table.uuid('payment_pointer_id').notNullable()
    table.foreign('payment_pointer_id').references('payment_pointers.id')
   
    table.string('asset_code').notNullable()
    table.float('value').notNullable()
    table.enum('type', ['INCOMING', 'OUTGOING']).notNullable()
    table.enum('status', ['PENDING', 'COMPLETED', 'REJECTED']).notNullable()

    table.timestamps(false, true, true)
```

File: packages/backend/src/index.ts [hunk 8]
```diff
@@ -1,31 +1,28 @@
...
-router.get(
-  '/',
-  async (_req: Request, res: Response<CustomResponse>, _next: NextFunction) => {
-    res.json({ message: '🐈' })
-  }
-)
+app.use(passport.initialize())
 
-app.use(router)
+app.use(mainRouter)
```
Comment (raducristianpopa) [inline packages/backend/src/index.ts:24]:
```suggestion
```

Let's comment/remove this line for now.

**Review by raducristianpopa — COMMENTED** [review]
I would suggest switching to camel case (`paymentPointerId`) for table columns.

File: docker/dev/docker-compose.yml [hunk 9]
```diff
@@ -6,11 +6,13 @@ services:
     environment:
       POSTGRES_USER: postgres
       POSTGRES_PASSWORD: password
+      POSTGRES_DB: testnet
     ports:
-      - '5432:5432'
+      - '5433:5432'
```
[the remark this answers is not in the document]
Comment (domin191013) [reply docker/dev/docker-compose.yml:11]: Well, rafiki is using port 5432, and 'docker compose up' wasn't working while rafiki project was running.

[the remark this answers is not in the document]
Comment (raducristianpopa) [reply docker/dev/docker-compose.yml:11]: Let's keep things as is until Rafiki comes into play.

[the remark this answers is not in the document]
Comment (domin191013) [reply docker/dev/docker-compose.yml:11]: But in my pc, rafiki is always running. And in future, our backend will call apis on rafiki? In this case, it'd be better to use different port, I'm not sure.
How do you think, Radu?

[the remark this answers is not in the document]
Comment (domin191013) [reply docker/dev/docker-compose.yml:11]: @raducristianpopa

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**